### ISSUE TITLE
UnSample() can't handle a length-one vector

### ISSUE THREAD

**Issue by sckaiser** [sckaiser/BlindDate#42]
for example:

> is.datechar("2015-1-31")
> Error: sample.sz > end.sz is not TRUE
> traceback()
> 4: stop(sprintf(ngettext(length(r), "%s is not TRUE", "%s are not all TRUE"), 
>        ch), call. = FALSE, domain = NA)
> 3: stopifnot(length(x) >= sample.sz, sample.sz > end.sz) at UnSample.R#15
> 2: UnSample(x, sample.sz) at is.datechar.R#12
> 1: is.datechar("2015-1-31")

**Comment by sckaiser** [comment]
Test cases:
```
UnSample(1, 1)
UnSample(1:100, 1)
UnSample(1:100, 4)
UnSample(1:100, 5)
UnSample(1:100, 9)
UnSample(1:100, 10)
UnSample(1:1000, 100)
```

**Comment by sckaiser** [comment]
Fixed via a161a7235469807067c8dcaf3d842d0e835def96. Note we use -1 as the end.sz default. While unconventional, in this case it lets us keep the stopifnot() clause sample size > end size.
